### DISCUSSION TITLE
Trying to add my 1st project to a Github repo

### DISCUSSION THREAD

**Discussion by Paul-Daniel-dev** [community/community#128717] [New to GitHub]
Hi there, I just finished my 1st web app project and im trying to put it into a repo I made on my account. Ive looked through and watched several sources that explain how this is done. I tried using git commands in my console but this also didnt work. I see now that I have a file (@next/swc-win32-x64-msvc) bigger than 100mb im trying to upload. My previous tries this was not an issue (maybe it was but it wasnt the error messages I was getting). I tried using git lfs to upload it and while following the steps on the official site https://git-lfs.com/ At step 3 I get told that there is not file im trying to use in the command: git add @next/swc-win32-x64-msvc (Ive also tried swc-win32-x64-msvc and swc-win32-x64-msvc.NODE) but this also has been to no use. Is there anything else I can do or is this project just completely unable to be put onto github? Thanks in advance for any help anyone can provide!

**Comment by Paul-Daniel-dev** [comment]
If any further information is needed im happy to provide it.

**Comment by arudeep15** [comment]


**Reply by Paul-Daniel-dev** [reply]
Im not quite sure what this means?

**Comment by IshworTM** [comment]
Hmm, I suggest you to try installing git bash on your windows from [here](https://www.git-scm.com/download/win) if that's complicated, open up your visual studio code and move to the directory that includes your project. And then follow the following instructions:

First of all open the repository your made and then copy the **SSH** link, then you can go back to your vs-code terminal and then execute the following commands:
`git init` - To initialise an empty git repository so that you can connect it to a remote one.
`git remote add origin git@github.com:YourRepository's Link.git` - To connect your local empty repository to the remote repository.
`git branch -M main` - To add and move to a **main** branch, it will be **master** by default. 
`git add <filename>` - To add a specific file so that you can push it to your GitHub repository.
`git add .` - To add all the files on your current working directory to your GitHub repository.
`git commit -m "your commit message"` - To commit your changes so that it can be reflected when you push the changes to GitHub.
`git push -u origin main` - To finally push your changes to your GitHub repository.

After following these instructions, you can check your GitHub repository, all the files you added should be shown there. Hope this helps. :)
Here's a [YouTube tutorial](https://www.youtube.com/watch?v=rLkz5UzAX_8&t=99s&pp=ygUWaXNod29yIGdpdGh1YiB0dXRvcmlhbA%3D%3D) that might somewhat be of use to you.

Really hope this helps you with your issue🥇:)

**Comment by Lakshan-Madushanka** [comment]
This could be a third-party package. In JavaScript, @ is often an alias for the root folder of the third-party packages repository, such as node_modules or node. The exact folder depends on the package manager you use (for npm, the folder is node_modules).

Typically, we don't submit these files to the GitHub repository because we can locally install them using the package.json file (the file name might vary depending on the package manager). This is one of the main advantages of using a package manager.

GitHub has a maximum file upload limit of 50 MB. Without third-party packages, this size is usually more than enough for any kind of project.

Therefore, I suggest creating a .gitignore file to instruct GitHub to ignore certain files. Add the following to your .gitignore file:

```
// .gitignore

/node_modules
````
Add the paths of all third-party library root folders or any other files that don't need to be uploaded to GitHub to your .gitignore file.

```
git add .gitignore
```
```
git commit -m 'Add .gitignore file'
```

If you complete this process correctly, the upload size to GitHub will be significantly reduced (likely less than 50 MB), making it manageable to upload to GitHub.

Just make sure to include right paths to .gitignore file.

You can learn more about .gitignore [here](https://git-scm.com/docs/gitignore).
